src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -204,9 +205,21 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     rawAccount: IAccountRaw,
     idx: number,
     globalIdx: number,
-    seed: Uint8Array,
+    seed?: Uint8Array,
   ): IAccount | null {
-    if (rawAccount.type !== ACCOUNT_TYPES.hdWallet) {
+    if (rawAccount.type === ACCOUNT_TYPES.privateKey && rawAccount.privateKey) {
+      const address = toChecksumAddress(privateKeyToAddress(rawAccount.privateKey).toString());
+      return {
+        idx,
+        globalIdx,
+        secretKey: rawAccount.privateKey!,
+        ...rawAccount,
+        privateKey: undefined,
+        address,
+        publicKey: Buffer.from(privateKeyToPublicKey(rawAccount.privateKey, false)),
+      } as IAccount;
+    }
+    if (rawAccount.type !== ACCOUNT_TYPES.hdWallet || !seed) {
```

Comment:
This test makes no sense here. `null` should be the default return value at the end of the method.  And each other case should be handled before. For example with the use of `switch..case`.